src/hooks/swap/useSyncTokenDefaults.ts

```diff
@@ -72,13 +72,34 @@ export default function useSyncTokenDefaults({
     updateSwap((swap) => ({ ...swap, ...defaultSwapState }))
   }, [defaultInputAmount, defaultInputToken, defaultOutputAmount, defaultOutputToken, updateSwap])
 
-  const lastChainId = useRef<number | undefined>(undefined)
   const isTokenListLoaded = useIsTokenListLoaded()
+  const lastChainId = useRef<number | undefined>(undefined)
```

Comment:
I don't think you need to use this `useRef` pattern at all. (aside: I'm updating it to use `useLast`, and making that more performant, see https://github.com/Uniswap/widgets/pull/56.).

Instead, you can set an effect over all of the props. This is probably easiest by not restructuring them to begin with.

    useEffect(() => {
      if (!isTokenListLoaded) return
      setToDefaults()
    }, [chainId, props, setToDefaults])